proto/arkeo/arkeo/keeper.proto

```diff
@@ -22,20 +22,43 @@ message Provider {
   ProviderStatus status = 5;
   int64 min_contract_duration = 6;
   int64 max_contract_duration = 7;
-  repeated cosmos.base.v1beta1.Coin subscription_rate = 8 [ (gogoproto.nullable) = false ];
-  repeated cosmos.base.v1beta1.Coin pay_as_you_go_rate = 9 [ (gogoproto.nullable) = false ];
-  string bond = 10 [
+  repeated ContractRate rates = 8;
+  string bond = 9 [
     (cosmos_proto.scalar) = "cosmos.Int",
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
     (gogoproto.nullable) = false
   ];
-  int64 last_update = 11;
-  int64 settlement_duration = 12;
+  int64 last_update = 10;
+  int64 settlement_duration = 11;
 }
 
-enum ContractType {
-  SUBSCRIPTION = 0;
-  PAY_AS_YOU_GO = 1;
+enum UserType {
+  SINGLE_USER = 0;
+  MULTI_USER = 1;
+}
+
+enum MeterType {
+  PAY_PER_BLOCK = 0; // previously called a Subscription
+  PAY_PER_CALL = 1;  // previously called Pay as you go 
+}
+
+message ContractRate {
+  MeterType meter_type = 1;
+  UserType user_type = 2;
```

Comment:
Disagree with this comment, its important to allow providers to decide what options they want to offer, by not providing a rate they don't provide that type of service.  Additionally, providers may want to charge more for different services.

Reply:
I feel like we're on a slippery slope here. Because we're complicating the service, we're also complicating the pricing which in turn, creates its own series of problems and downstream affects. The original design was very clean and simple, and we're taking it now in a different direction where things are getting muddy for a feature we all admit will get abused. Will need to think more on this, feels like a dangerous road to me.

Reply:
I thought we had already moved past this conversation.  I am not willing to spend any more time debating the merit of this feature as we have talked through these points at length in several different mediums.  We need to move forward.